src/status_im/native_module/impl/module.cljs

```diff
@@ -162,6 +162,51 @@
                            (callback r')))]
            (.callJail status jail-id (types/clj->json path) (types/clj->json params') cb))))))
 
+;; TODO(rasom): temporal solution, should be fixed on status-go side
+(def check-raw-calls-interval 400)
+(def interval-between-calls 100)
+;; contains all calls to jail before with duplicates
+(def raw-jail-calls (atom '()))
+;; contains only calls that passed duplication check
+(def jail-calls (atom '()))
+
+(defn remove-duplicate-calls
+  "Removes duplicates by [jail path] keys, remains the last one."
+  [[all-keys calls] {:keys [jail-id path] :as call}]
+  (if (contains? all-keys [jail-id path])
+    [all-keys calls]
+    [(conj all-keys [jail-id path])
+     (conj calls call)]))
+
+(defn check-raw-calls-loop!
+  "Only the last call with [jail path] key is added to jail-calls list
+  from raw-jail-calls"
+  []
+  (go-loop [_ nil]
+    (let [[_ new-calls] (reduce remove-duplicate-calls [#{} '()] @raw-jail-calls)]
+      (reset! raw-jail-calls '())
+      (swap! jail-calls (fn [old-calls]
+                           (concat new-calls old-calls))))
+    (recur (<! (timeout check-raw-calls-interval)))))
+
+(defn execute-calls-loop!
+  "Calls to jail are executed ne by one with interval-between-calls,
+  which reduces chances of response shuffling"
+  []
+  (go-loop [_ nil]
+    (let [next-call (last @jail-calls)]
+      (swap! jail-calls butlast)
+      (when next-call
+        (execute-call next-call)))
```

Comment:
Are we worried about an exceptional call killing the go-loop?

Reply:
I don't think so, and anyway if error will happen there it will not pass silently, we will have red screen in dev mode and crash in prod. Like a regular code.